README.md

```diff
@@ -2,4 +2,160 @@
+
+### Configuration file
+
+Kombucha’s behavior is governed by a declarative JSON configuration file. This section explains the contents of that file in detail, or you can [jump into the repo](blob/master/sample.json) and check out a sample configuration file directly.
+
+The top level of the configuration file looks like this:
+```json
+{
+  "userAgent": "kombucha-this-is-a-test-v-0.0001",
+  "snaps": [ … ]
+}
+```
+* `userAgent` is the User Agent string you’d like Kombucha to use when it issues HTTP requests on your behalf
+* `snaps` is an array of API endpoints that you would like tested. Here’s what a single entry in the `snaps` array looks like:
+```json
+{
+  …
+  "snaps": [
+    {
+      "host": "httpbin.org",
+      "path": "/response-headers",
+      "queryItems": {
+        "foo": "1",
+        "bar": "2",
+        "baz": "3"
+      },
+      "scheme": "https",
+      "__snapType": "__GET"
+    },
+  …
+  ]
+}
+```
+* `__snapType` specifies the type of request Kombucha is going to issue. `__GET` is a simple HTTP `GET` request, and the remaining parameters are interpreted accordingly: `queryItems` are converted into key-value pairs and appended onto the `path` specified, etc.
+   * The `snaps` entry above results in an HTTP `GET` to `https://httpbin.org/response-headers?foo=1&bar=2&baz=3`. In addition, Kombucha always sends an `Accept` header of `application/json`.
+* Kombucha also supports a `__GRAPHQL` `snapType` for testing GraphQL endpoints. Documentation for this is forthcoming. 😄
+
+#### Customizing output
+
+When executing a snapshot test, Kombucha converts both the snapshot JSON and the “work” (live response) JSON into data in memory. It then executes what we call [the `structure` check](#the-structure-check) to look for the API breakages and problems we’ve discussed above. 
+
+However, with the full JSON data in memory, we can perform other checks on the response as we work. It may be useful to think of this as a built-in “linter” for the response that runs alongside the core snapshotting feature that has previously been described.
+
+The user can add a `__preferences` blob to their configuration file to specify exactly the mixture of checks they’d like to run:
+```json
+{
+  …
+  "snaps": [
+    {
+      "host": "httpbin.org",
+      "path": "/response-headers",
+      "queryItems": {},
+      "scheme": "https",
+      "__preferences": {
+        "errors": [
+          "string-bools",
+          "string-numbers",
+          "structure"
+        ],
+        "infos": [
+          "flag-new-keys"
+        ],
+        "warnings": [
+          "array-consistency",
+          "empty-arrays",
+          "empty-objects"
+        ]
+      },
+      "__snapType": "__GET"
+    },
+  …
+  ]
+}
+```
+* the `__preferences` above specify that, in addition to `structure`, the checks named `string-bools` and `string-numbers` should also be run, and if they produce output, that output should be treated as errors, failing the test run. 
```

Comment:
? - `__preferences` specify ....
For consistency with the rest of the doc and how parameters are documented.